plugin/core/transports.py

```diff
@@ -179,8 +179,18 @@ def _stderr_loop(self) -> None:
 def create_transport(config: ClientConfig, cwd: Optional[str], window: sublime.Window,
                      callback_object: TransportCallbacks, variables: Dict[str, str]) -> JsonRpcTransport:
     tcp_port = None  # type: Optional[int]
+    listener_socket = None  # type: Optional[socket.socket]
     if config.tcp_port is not None:
-        tcp_port = _find_free_port() if config.tcp_port == 0 else config.tcp_port
+        # < 0 means we're hosting a TCP server
+        if config.tcp_port < 0:
```

Comment:
This condition (together with -1 check below) seems to suggest that other negative values apart from -1 are also supported. Is that the intention?

Reply:
Never mind, I've read the comment now.